packages/components/checkbox/src/Checkbox.test.tsx

```diff
@@ -229,6 +240,26 @@ describe('CheckboxGroup', () => {
     expect(screen.getByRole('checkbox', { name: 'My accessible field label' })).toBeInTheDocument()
   })
 
+  it('should handle the reverse prop', () => {
+    const { container, rerender } = render(
+      <CheckboxGroup>
+        <Checkbox value="hello">hello</Checkbox>
+      </CheckboxGroup>
+    )
+    const label = screen.getByText('hello')
+
+    const getFirstRelevantElement = () => container.firstChild?.firstChild?.firstChild
```

Comment:
Those sequences of chaining `firstChild` are confusing I think. I don't really understand what it tested. You can use this trick to get the label by it's text + the checkbox by it's accessible name and compare DOM order:

```
it.only('should handle the reverse prop', () => {
    const { rerender } = render(
      <CheckboxGroup>
        <Checkbox value="hello">Hello</Checkbox>
      </CheckboxGroup>
    )

    const hasLabelBeforeInput = (accessibleName: string) => {
      const checkboxLabel = screen.getByText(accessibleName)
      const checkboxElement = screen.getByRole('checkbox', { name: accessibleName })

      return !!(
        checkboxElement.compareDocumentPosition(checkboxLabel) & Node.DOCUMENT_POSITION_FOLLOWING
      )
    }

    expect(hasLabelBeforeInput('Hello')).toBeTruthy()

    rerender(
      <CheckboxGroup reverse>
        <Checkbox value="hello">Hello</Checkbox>
      </CheckboxGroup>
    )

    expect(hasLabelBeforeInput('Hello')).toBeFalsy()
  })
```